File: sol/2559/2559.cpp
struct Solution {
    inline bool isVowel(char c){
        switch (c){
            case 'a':
            case 'e':
            case 'i':
            case 'o':
            case 'u':
                return true;
            default:
                return false;
        }
    }

    vector<int> vowelStrings(vector<string>& words, vector<vector<int>>& queries) {
        int n = words.size(), q = queries.size();
        vector<int> count(n + 1, 0), res(q);
        for (int i = 0; i < n; ++i){
            count[i + 1] = count[i] + (this->isVowel(words[i].front()) && this->isVowel(words[i].back()));
        }
        for (int i = 0; i < q; ++i){
            res[i] = count[queries[i].back() + 1] - count[queries[i].front()];
        }
        return res;
    }
};

File: sol/2559/2559.py
class Solution:
    def vowelStrings(self, words: List[str], queries: List[List[int]]) -> List[int]:
        vowels = frozenset('aeiou')
        n = len(words)
        vowelStrCount = [0] * (n + 1)  # [0:i)
        for i in range(n):
            vowelStrCount[i + 1] = vowelStrCount[i] + int((words[i][0] in vowels) and (words[i][-1] in vowels))
        return [vowelStrCount[b + 1] - vowelStrCount[a] for a, b in queries]
